src/js/components/UserAccountDropdownTrigger.js

```diff
@@ -23,48 +23,22 @@ class UserAccountDropdownTrigger extends mixin(StoreMixin) {
   }
 
   render() {
-    let { primaryContent, secondaryContent = null } = this.props;
-
-    // Promote secondary content to primary content in the event that secondary
-    // is the only available content.
-    if (!primaryContent && secondaryContent) {
-      primaryContent = secondaryContent;
-      secondaryContent = null;
-    } else if (secondaryContent) {
-      secondaryContent = (
-        <div className="header-subtitle">{secondaryContent}</div>
-      );
-    }
+    const { content, onTrigger } = this.props;
 
     return (
-      <header className="header" onClick={this.props.onTrigger}>
-        <a className="header-dropdown">
-          <div className="header-content">
-            <div className="header-image-wrapper">
-              <div className="header-image" />
-            </div>
-            <div className="header-details">
-              <span className="header-title">
-                <span>{primaryContent}</span>
-              </span>
-              {secondaryContent}
-            </div>
-          </div>
-        </a>
-      </header>
+      <span className="header-bar-dropdown-trigger" onClick={onTrigger}>
+        <span className="header-bar-dropdown-trigger-content text-overflow">
+          {content}
+        </span>
+      </span>
     );
   }
 }
 
-UserAccountDropdownTrigger.defaultProps = {
-  showCaret: false,
-  onTrigger() {}
-};
-
 UserAccountDropdownTrigger.propTypes = {
-  clusterName: PropTypes.node,
+  content: PropTypes.string.isRequired,
   onUpdate: PropTypes.func,
-  showCaret: PropTypes.bool
+  onTrigger: PropTypes.func
```

Comment:
usually we go with either `isRequired` or a default value for functions.

Reply:
For this prop, it is not required. I don't see a default value to be set here.

Reply:
I disagree, a trigger without such a function wouldn't make any sense and would be conceptually probably a misused, still I think this is not so important right now

Reply:
I agree with @DanielMSchmidt here but not a blocker